### PR TITLE
Fix the case when reindex failed and catalog_product_price has a Reindex required status

### LINKED ISSUE #35796: Magento unable to reindex bundle product not assigned to any sait (DisabledProductOptionPriceModifier.php on line 117).
**Preconditions**
Install Magento 2.4-develop with sample data.
 
**Issue** 
product price index (catalog*product*price) not working if bundle product not assigned to any website 
 
**Steps To Reproduce**
1)Go to admin ->catalog ->products -> create new bundle Or edit existing bundle product 
2)"Product in Websites" Tab  -> uncheck all checkboxes (website) and save the product 
3)Perform reindex 

 
**Expected result**
Reindex should be performed successfully 
 
**Actual result**
We get the below error while running the product price index
 
```java
Warning: Undefined array key <bundel product id > in vendor/magento/module-bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php on line 117
```

### PR DESCRIPTION
**Pull request by artskirk**
### Description (*)

Added the additional undefined index checking to prevent the following error:
```
bin/magento indexer:reindex catalog_product_price
Product Price index process error during indexation process:
Warning: Undefined array key 45 in /var/www/html/m2d/app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php on line 124
```

### Fixed Issues (if relevant)

1. Fixes magento/magento2#35796

### Manual testing scenarios (*)

1. Create bundle product
2. In bundle product go to tab "Product in Websites" & uncheck the Websites.
3. Click save & close.
4. In CLI enter bin/magento indexer:reindex catalog_product_price

**Expected result:** Reindex success and catalog_product_price have a status ready.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ private function getWebsiteIdsOfProduct(int $entityId): array
         foreach ($connection->fetchCol($select) as $websiteId) {
             $this->websiteIdsOfProduct[$entityId][] = (int)$websiteId;
         }
```
Comment (ihor-sviziev) [inline app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php:124]: Looks like the issue coming from not declaring the empty array for the entity id. I would suggest adding it.
```suggestion
        $this->websiteIdsOfProduct[$entityId] = [];
        foreach ($connection->fetchCol($select) as $websiteId) {
            $this->websiteIdsOfProduct[$entityId][] = (int)$websiteId;
        }
```

File: app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php [hunk 2]
```diff
@@ -121,7 +121,7 @@ private function getWebsiteIdsOfProduct(int $entityId): array
...
-        return $this->websiteIdsOfProduct[$entityId];
+        return $this->websiteIdsOfProduct[$entityId] ?? [];
```
Comment (ihor-sviziev) [inline app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php]:
```suggestion
        return $this->websiteIdsOfProduct[$entityId];
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
Hi @artskirk,
Could you please review my suggestion and add test coverage for this case?

**Comment by artskirk** [comment]
> Hi @artskirk, Could you please review my suggestion and add test coverage for this case?

yes, I will

**Comment by artskirk** [comment]
@ihor-sviziev I have checked your suggested solution, it's work as expected. I will cover it with unit tests.

**Comment by ihor-sviziev** [comment]
@artskirk, please make sure that the unit test you wrote will fail w/o your changes and will pass after.

**Comment by artskirk** [comment]
@magento run Unit Tests

**Comment by ishakhsuvarov** [comment]
Tests have not been fixed for a long time, hence closing this PR for now.

Please feel free to reopen if you wish to revisit this.

**Pull request closed without merging**